### PR TITLE
Refactor the Louisiana exempt income variable to use the federal tax deduction

### LINKED ISSUE #4462: Refactor Louisiana exempt income formula
The formula for the `la_agi_exempt_income` variable is too long and has dubious variable names, making it difficult to tell whether or not the logic in the formula is correct or not.   The formula needs to be broken up making several of the formula's local variables into global variables.

An example of the confusion is this local variable:

https://github.com/PolicyEngine/policyengine-us/blob/cf9237a7c8e14674ebd2e38e50f2afeff24362c0/policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py#L30-L33

.

I don't understand how `AGI-itemized_deduction-income_tax_before_credits` can be an income tax variable.   Perhaps it is, but the logic is not transparent.

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #4462

### CODE REVIEW

File: policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py [hunk 1]
```diff
@@ -6,6 +6,7 @@ class la_agi_exempt_income(Variable):
     entity = TaxUnit
     label = "Louisiana income that is exempt from the adjusted gross income"
     defined_for = StateCode.LA
+    reference = "https://revenue.louisiana.gov/TaxForms/IT540i(2021)%20Instructions.pdfpage=9"
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py:9]: can you add a legal code reference?

File: policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py [hunk 2]
```diff
@@ -14,32 +15,31 @@ def formula(tax_unit, period, parameters):
...
-            income_tax_present = pre_exempt_income_tax > 0
-            exempt_income_tax_reduction = (
-                pre_exempt_income_tax * exempt_income_rate
+            federal_tax_deduction_present = federal_tax_deduction > 0
+            # Multiply the federal tax deduction by the exempt income rate
+            reduced_federal_tax_deduction = (
+                federal_tax_deduction * exempt_income_rate
             )
-            # The smaller of the two options is applied
+            # The smaller of the two options is applied if taxable income is above 0
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py:35]:
```suggestion
            # The smaller of the two options is applied if taxable income exceeds zero
```

File: policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py (hunk 1 above)
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py:9]: this link doesn't work for me

File: policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py [hunk 3]
```diff
@@ -14,32 +15,31 @@ def formula(tax_unit, period, parameters):
...
-            pre_exempt_income_tax = max_(
-                agi - claimed_itemized_deductions - fed_tax_deduction, 0
+            # The second option only applies if the tax unit has a federal tax deduction
+            federal_tax_deduction = tax_unit(
+                "la_federal_tax_deduction", period
             )
-            income_tax_present = pre_exempt_income_tax > 0
-            exempt_income_tax_reduction = (
-                pre_exempt_income_tax * exempt_income_rate
+            federal_tax_deduction_present = federal_tax_deduction > 0
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py:30]: or _eligible if it's not about claiming
```suggestion
            claims_federal_tax_deduction = federal_tax_deduction > 0
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py (hunk 1 above)
Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py]: The legal code section has been repealed and I cannot find the text pre 2022

File: policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py [hunk 4]
```diff
@@ -6,40 +6,39 @@ class la_agi_exempt_income(Variable):
...
             exempt_income_rate = np.zeros_like(agi)
             mask = agi != 0
             exempt_income_rate[mask] = total_exempt_income[mask] / agi[mask]
-            itemizes = tax_unit("tax_unit_itemizes", period)
-            itemized_deductions = tax_unit("la_itemized_deductions", period)
-            claimed_itemized_deductions = itemizes * itemized_deductions
-            fed_tax_deduction = tax_unit("la_federal_tax_deduction", period)
-            pre_exempt_income_tax = max_(
-                agi - claimed_itemized_deductions - fed_tax_deduction, 0
+            # The second option only applies if the tax unit has a federal tax deduction
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py:25]: Could you spell this out more? I see it references instructions for Line 9, is that where this is coming from? Line and page numbers may help.

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py:25]: Lmk what you think

File: policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py [hunk 5]
```diff
@@ -6,40 +6,43 @@ class la_agi_exempt_income(Variable):
     entity = TaxUnit
     label = "Louisiana income that is exempt from the adjusted gross income"
     defined_for = StateCode.LA
-    unit = USD
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py:9]: this shouldn't be removed right?

File: policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py [hunk 6]
```diff
@@ -6,40 +6,43 @@ class la_agi_exempt_income(Variable):
...
             exempt_income_rate[mask] = total_exempt_income[mask] / agi[mask]
-            itemizes = tax_unit("tax_unit_itemizes", period)
-            itemized_deductions = tax_unit("la_itemized_deductions", period)
-            claimed_itemized_deductions = itemizes * itemized_deductions
-            fed_tax_deduction = tax_unit("la_federal_tax_deduction", period)
-            pre_exempt_income_tax = max_(
-                agi - claimed_itemized_deductions - fed_tax_deduction, 0
+            # The second option only applies if the tax unit has a federal tax deduction
+            # The step by step calculation is as follows:
+            # Line 1, representing federal AGI is devided by Line 4H
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py:27]:
```suggestion
            # Line 1, representing federal AGI is divided by Line 4H 
```

File: policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py [hunk 7]
```diff
@@ -6,40 +6,43 @@ class la_agi_exempt_income(Variable):
...
-            itemizes = tax_unit("tax_unit_itemizes", period)
-            itemized_deductions = tax_unit("la_itemized_deductions", period)
-            claimed_itemized_deductions = itemizes * itemized_deductions
-            fed_tax_deduction = tax_unit("la_federal_tax_deduction", period)
-            pre_exempt_income_tax = max_(
-                agi - claimed_itemized_deductions - fed_tax_deduction, 0
+            # The second option only applies if the tax unit has a federal tax deduction
+            # The step by step calculation is as follows:
+            # Line 1, representing federal AGI is devided by Line 4H 
+            # which is the total exempt income. This rate is multiplied by
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py:28]:
```suggestion
            # which is the total exempt income. This rate (exempt_income_rate) is multiplied by 
```

File: policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py [hunk 8]
```diff
@@ -6,40 +6,43 @@ class la_agi_exempt_income(Variable):
...
+            # Line 1, representing federal AGI is devided by Line 4H 
+            # which is the total exempt income. This rate is multiplied by 
+            # Line 9 of the Form IT-540, representing the federal tax deduction
+            federal_tax_deduction = tax_unit(
+                "la_federal_tax_deduction", period
             )
-            income_tax_present = pre_exempt_income_tax > 0
-            exempt_income_tax_reduction = (
-                pre_exempt_income_tax * exempt_income_rate
+            federal_tax_deduction_eligible = federal_tax_deduction > 0
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py:33]: can you move this below to align with the comment?

File: policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py [hunk 9]
```diff
@@ -6,40 +6,43 @@ class la_agi_exempt_income(Variable):
...
+            # Multiply the federal tax deduction by the exempt income rate
+            reduced_federal_tax_deduction = (
+                federal_tax_deduction * exempt_income_rate
             )
-            # The smaller of the two options is applied
+            # The smaller of the two options is applied if taxable income exceeds zero
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/la/tax/income/exempt_income/la_agi_exempt_income.py:38]: comment where on the form or instructions this is

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**